### PR TITLE
Fix Mobile not using ISpeedModifier properly

### LINKED ISSUE #8260: ISpeedModifier does not work anymore
Both `TakeCover` and `ScaredyCat` are affected. Tested on RA and TS.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/17621032-ispeedmodifier-does-not-work-anymore?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
Fixes #8260.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 1]
```diff
@@ -599,7 +599,7 @@ public int MovementSpeedForCell(Actor self, CPos cell)
 				return 0;
 
 			speed *= Info.Speed;
-			foreach (var t in speedModifiers)
+			foreach (var t in speedModifiers.Value)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: As we discussed in IRC we should clean up the easy `// TODO` six lines above while here.

**Comment by penev92** [comment]
Updated.

**Comment by obrakmann** [comment]
Fixes the issue for me. :+1:

**Comment by Mailaender** [comment]
Confirming the fix. :white_check_mark: Came up with similar code myself, but never got it finally. resolved. Thanks!

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/d859074b3b3dea486d181c6d2399d1b46e32d84a...9a89c2a379f6164e5a9f7d5cf0063f2ee42ae5b3)

**Pull request merged**